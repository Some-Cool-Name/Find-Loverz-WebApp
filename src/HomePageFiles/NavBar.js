import React from 'react';
import { Link, useHistory } from 'react-router-dom';
import { deleteFromStorage } from '../HelperClasses/StorageHandler';

export default function NavBar({ setUser, user}) {
    let history = useHistory();

    // handles user logout
    const handleLogout = () => {
        deleteFromStorage('user');
        setUser(null);
        history.push('/');
    }

    return (        
        <navbar>
            <div className="left-nav">
                <h1 id="nav-app-name">Find Loverz</h1>
                <div className="profile-container">
                    <div className = "match-profile-picture"><img src = {user[0]?.profile_picture} alt="" /> </div>
                        <div className="profile-name">
                            <p id="feed-username">{user[0]?.username}</p>                          
                        </div>
                </div>
            </div>

            <div className="right-nav">
                <Link to="/feed" style={{marginRight: 5}} ><p>Home</p></Link>
                <Link to="/matches" style={{marginRight: 5}} ><p>Chats</p></Link>
                <Link to="/profile" style={{marginRight: 5}} ><p>Profile</p></Link>
                <input type="text" className="nav-search-bar" id="nav-search-bar" placeholder="Search"></input>
                <div className="search-button"><i class="uil uil-search"></i></div>
                <Link to="/" onClick={handleLogout} ><p>Logout</p></Link>
            </div>
        </navbar>
    )
}
