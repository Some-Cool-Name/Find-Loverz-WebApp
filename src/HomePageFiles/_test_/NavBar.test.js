import {render, fireEvent} from "@testing-library/react";
import NavBar from "../NavBar.js"
import { StaticRouter } from 'react-router';
import { Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
//

test("testing if it renders and buttons", () =>{
    
    const component = renderer.create(
        <StaticRouter location="../.js" context={<NavBar />}>
            <Link to="../Profile.js">profile</Link>
        </StaticRouter>

    
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

