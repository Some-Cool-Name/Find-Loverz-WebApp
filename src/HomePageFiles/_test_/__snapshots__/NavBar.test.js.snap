// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing if it renders and buttons 1`] = `
<a
  href="/../../Profile.js"
  onClick={[Function]}
>
  profile
</a>
`;
